03/Vector3.cpp

```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+#include "Vector3.h"
+
+
+Vector3::Vector3(int x, int y)
+	: mX(x)
+	, mY(y)
+{
+}
+
+int Vector3::Multiply(const Vector3& reference) const
```

Comment:
your functions in vector classes look all wrong so i'm goign to writ edown what we want here


Function version
1. void Scale(const Vector3&);
2. void Scale(float s);
3. Vector3 Multiply(const Vector3&);  // do memberwise multiply
4. Vector3 Mutiply(float s);

operator version
1. void operator*=(const Vector3&);
2. void operator*=(float s);
3. Vector3 operator*(const Vector3&);  // do memberwise multiply
4. Vector3 operator*(float s);

Reply:
Just to be clear..

1. Scale function does multiplying 2 power n to member variables(mX, mY) and returns the result
2 Multiply function does multiplying the member variables to another object's member variables and returns the result

Am I right?

Reply:
1. scale doesn't return anything it's void so it changes the member variables

mX *= s;
mY *= s;

would do

actually change float s to int s

2. doesn't change member variable at all

when you do 

3 * 5 = 15

3 and 5 doesn't change.

Reply:
No

scale doesn't return anything it's void so it changes the member variables
mX *= s;
mY *= s;

would do

actually change float s to int s

doesn't change member variable at all
when you do

3 * 5 = 15

3 and 5 doesn't change.